### PR TITLE
Proper in-place algebra

### PR DESCRIPTION
**Pull request by evgueni-ovtchinnikov**
In-place data containers algebra (`+=`, `-=`, `*=` and `/=`) was temporarily implemented in SIRF via numpy in-place algebra and `as_array`/`fill`.

This PR replaces the temproary implementation with a proper one that eliminates unnecessary copying of data between SIRF data containers and numpy arrays.

### CODE REVIEW

File: examples/Python/PET/acquisition_data.py [hunk 1]
```diff
@@ -100,12 +100,16 @@ def main():
     print('norm of acq_data.as_array(): %f' % numpy.linalg.norm(acq_array))
     print('acq_data.norm(): %f' % s)
     print('sqrt(acq_data.dot(acq_data)): %f' % math.sqrt(t))
-    diff = new_acq_data - acq_data
+    diff = new_acq_data
+    diff -= acq_data
```
Comment (KrisThielemans) [inline examples/Python/PET/acquisition_data.py:104]: this update looks a non-intuitive way to write the code. why would we do this?

File: src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp [hunk 2]
```diff
@@ -378,11 +378,13 @@ complex_float_t a, complex_float_t b)
 	int n = y.number();
 	ISMRMRD::Acquisition ax;
 	ISMRMRD::Acquisition ay;
+	ISMRMRD::Acquisition acq;
```
Comment (KrisThielemans) [inline src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp:381]: updates here don't seem have anything to do with the title of this PR. Are these intentional?

File: src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h [hunk 3]
```diff
@@ -222,6 +222,7 @@ namespace sirf {
 		// abstract methods
 
 		virtual void empty() = 0;
+		virtual void take_over(MRAcquisitionData&) = 0;
```
Comment (KrisThielemans) [inline src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h:225]: not related to PR?

File: examples/Python/PET/acquisition_data.py (hunk 1 above)
Comment (evgueni-ovtchinnikov) [reply examples/Python/PET/acquisition_data.py:104]: just was checking `-=`

File: src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp (hunk 2 above)
Comment (evgueni-ovtchinnikov) [reply src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp:381]: all updates are necessary for in-place algebra the way I implemented it: e.g. now that we want to overwrite acquisition data, we need to check first whether the acquisition to be overwritten is not to be ignored

File: src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h (hunk 3 above)
Comment (evgueni-ovtchinnikov) [reply src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h:225]: directly related - actually, the main difference from the previous approach is that now new data is created before the old one is dismissed, after which the object takes the ownership of the new data and discards the old one

**Pull request merged**